### ISSUE TITLE
Source position may be modified

### ISSUE THREAD

**Issue by fight4dream** [ExtendRealityLtd/Tilia.Trackers.PseudoBody.Unity#153]
### Environment

* Code source: tilia psuedo body
* Code version:  36fdc67b7fbeb639779fd5138e2685396b0df574
* Platform version: Unity 2019.4.17f1

### Steps to reproduce

1. new scene, add TrackedAlias and any camera rigs
2. create empty child under HeadsetAlias, name it HeadJoint, position (0, -0.1, -0.12)
3. create a cube, name it Floor, position (0, -5, 0), scale (10, 10, 10)
4. create a cube, name it Lava, set layer Water, position (0, -7, 0), scale (10, 10, 10), use a material to make it red
5. add PseudoBody, set Source = HeadJoint, Offset = PlayAreaAlias
![image](https://user-images.githubusercontent.com/22952143/146041616-8dab4364-0969-492f-b3a0-e01abd6468b2.png)
6. in PseudoBody/CollidableVolume:CollisionTracker.CollisionStarted, we'll make it detect collision with Lava
![image](https://user-images.githubusercontent.com/22952143/146041458-b962e5fe-92ab-4f19-9e69-8fbbdd354366.png)
7. the screen caps that follows are equivalent to the pseudo code: 
```
public void ResetIfLava(...)
{
    if (collided with Lava)
    {
        PseudoBody.SetActive(false);
        Vector3 position = PlayAreaAlias.transform.position;
        position.y = 0;
        PlayAreaAlias.transform.position = position;
        Floor.SetActive(true);
        yield return new WaitForEndOfFrame();
        PseudoBody.SetActive(true);
    }
}
```
![image](https://user-images.githubusercontent.com/22952143/146042990-5a14869f-9c14-4c56-b337-748337550ca4.png)
![image](https://user-images.githubusercontent.com/22952143/146043009-95d4bf54-d960-48a2-bcbd-fd97284c3856.png)
![image](https://user-images.githubusercontent.com/22952143/146043067-1ab8e92e-6af0-4735-82b6-9071468d251e.png)
![image](https://user-images.githubusercontent.com/22952143/146043091-9c34c905-2152-4eab-93df-a8cccc937740.png)
![image](https://user-images.githubusercontent.com/22952143/146043115-d47d7db9-5e44-436d-8e4a-e85dcfdc4673.png)

8. play the scene, deactivate Floor, player/pseudobody should fall onto Lava, and it will trigger the above procedure to teleport the player back to the Floor

### Expected behavior

After the teleport, the HeadJoint should still be at position (0, -0.1, -0.12).

### Current behavior

After the teleport, the HeadJoint is moved.
![image](https://user-images.githubusercontent.com/22952143/146045338-53436645-af49-43bd-b696-77fc58ddfccb.png)

**Comment by thestonefox** [comment]
What if the setup doesn't even have the psuedobody in play?

The headjoint is part of the tracked alias, and the movement is being caused by the teleporter.

The pseudobody code does not alter the position of the source, only the character controller (as far as I can see).

I'd recommend simplifying the steps to reproduce as there could be a number of points where an issue is being introduced by the steps involved right now.

**Comment by fight4dream** [comment]
1. new scene, add TrackedAlias and any camera rigs
2. create empty child under HeadsetAlias, name it HeadJoint, position (0, -0.1, -0.12)
3. the screen caps that follow are equivalent to the pseudo code:
```public void ResetPlayArea(...)
{
        Vector3 position = PlayAreaAlias.transform.position;
        position.y = 0;
        PlayAreaAlias.transform.position = position;
}
```
![image](https://user-images.githubusercontent.com/22952143/146768923-c0f94899-37c7-481c-af42-d0f444e555e0.png)
4. add a Input manager button action to make the teleport happen when 'spacebar' is pressed
![image](https://user-images.githubusercontent.com/22952143/146769168-a293cb24-e03d-4368-9bb8-ba529d9322fb.png)
5. play the scene, drag the PlayAreaAlias down to simulate the original pseudo body falling down, observe the HeadJoint is still at position (0, -0.1, -0.12).
6. press 'spacebar' to teleport. Observe the HeadJoint is still at position (0, -0.1, -0.12).

cannot reproduce the HeadJoint position modified issue with these steps.

@thestonefox please advise for the next test.

**Comment by fight4dream** [comment]
1. new scene, add TrackedAlias and any camera rigs
2. create empty child under HeadsetAlias, name it HeadJoint, position (0, -0.1, -0.12)
3. create a cube, name it Floor, position (0, -5, 0), scale (10, 10, 10)
4. add PseudoBody, set Source = HeadJoint, Offset = PlayAreaAlias
5. play the scene, now drag the Floor down, PseudoBody (and playarea) should fall down onto new floor position.

Observe that the the HeadJoint position is modified. Also notice that its position may be constantly being updated due to some floating point error

**Comment by thestonefox** [comment]
yeah I can recreate with those, good spot on the processor line, i'll see whats going on

**Comment by thestonefox** [comment]
it's a bit odd that it changes the source position, because usually that is the headset alias which is being kept in sync with the actual headset, so changing the position of it doesn't really affect anything.

But obviously in this regard it is causing an issue. I'm not entirely sure why the source is changed at all.

**Comment by thestonefox** [comment]
What about something just like this?

https://github.com/ExtendRealityLtd/Tilia.Trackers.PseudoBody.Unity/pull/154

you can just untick the update source position bool?

**Comment by fight4dream** [comment]
confirmed #154 in this situation the source HeadJoint position is not modified.
however, in what situations do we need to modify the source position?

**Comment by thestonefox** [comment]
I don't know why the source position is modified, chris wrote the psuedo body stuff originally and I'm struggling to think of a time when you'd need to modify it. I remember there was something about you could use it without vr stuff and in those instances it probably makes sense.
